pyomo/contrib/cspline_external/cspline_parameters.py

```diff
@@ -313,3 +359,71 @@ def yxx_endpoint_eqn(blk, s):
         else:
             j = s
         return _fxx_cubic(m.x[j], m.alpha, s) == 0
+
+
+def add_decreasing_constraints(m, tol=0):
```

Comment:
These functions assume that `m` has a set `knt_idx`, and that they can create components with specific hardcoded names.  This feels like these should be methods on a custom Block class that disallows users to create components so you don't have to worry about conflicts and *know* that the required attributes are present.